test_unstructured/partition/pdf_image/test_pdf.py

```diff
@@ -1362,3 +1363,27 @@ def test_analysis_artifacts_saved():
         for el in expected_layouts:
             for page in expected_pages:
                 assert bboxes_dir / f"page{page}_layout_{el}.png" in bboxes_files
+
+
+@pytest.mark.parametrize(
+    ("max_pages", "expected_result"),
+    [
+        (3, "expected_successful"),
+        (2, "expected_error"),
+    ],
+)
+def test_max_pages_argument(max_pages, expected_result):
+    if expected_result == "expected_successful":
+        pdf.partition_pdf_or_image(
+            filename=example_doc_path("pdf/reliance.pdf"),
```

Comment:
one more would be nice and `filename` added as a separate `@pytest.mark.parametrize` variable, e.g. `example-docs/layout-parser-paper-with-empty-pages.pdf`